llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -6905,6 +6905,15 @@ static bool planContainsAdditionalSimplifications(VPlan &Plan,
       if (isa<VPPartialReductionRecipe>(&R))
         return true;
 
+      // The VPlan-based cost model can analysis if recipes is scalar
+      // recursively, but legacy cost model cannot.
+      if (auto *WidenMemR = dyn_cast<VPWidenMemoryRecipe>(&R);
+          WidenMemR &&
+          vputils::isSingleScalar(WidenMemR->getAddr()) !=
```

Comment:
Nit I don't think the `if (Foo = ...; Foo && ...)` pattern is used much in the loop vectorizer
```suggestion
      if (auto *WidenMemR = dyn_cast<VPWidenMemoryRecipe>(&R))
          if (vputils::isSingleScalar(WidenMemR->getAddr()) !=
```